lib/puppet/provider/package/dpkg.rb

```diff
@@ -94,6 +99,7 @@ def install
     args << '-i' << file
 
     dpkg(*args)
+    self.hold if @resource[:hold]
```

Comment:
Should we re-hold in an ensure block in case `dpkg` raises?

```ruby
self.unhold
begin
  if @resource[:configfiles] == :keep
  ...
  dpkg(*args)
ensure
  self.hold if @resource[:hold]
end
```

Reply:
did similar changes in all cases